examples/00-load/create-pointset.py

```diff
@@ -0,0 +1,54 @@
+"""
+.. _create_pointset_example:
+
+Create a PointSet
+~~~~~~~~~~~~~~~~~
+
+A :class:`pyvista.PointSet` is a concrete class representing a set of points
+that specifies the interface for datasets that explicitly use "point" arrays to
+represent geometry. This class is useful for improving the performance of
+filters on point clouds.
+
+This example shows the performacne improvement when cliping using a
+:class:`pyvista.PointSet` using the :func:`pyvista.DataSet.clip` filter.
+
+"""
+import time
+
+import pyvista as pv
+from pyvista import examples
+
+lidar = examples.download_lidar()
+
+tstart = time.time()
+clipped = lidar.clip(origin=(0, 0, 1.76e3), normal=(0, 0, 1))
+t_elapsed = time.time() - tstart
+print(f"Time to clip with a PolyData {t_elapsed:.2f} seconds.")
+
+###############################################################################
+# Plot the Clipped polydata
+clipped.plot(show_scalar_bar=False)
+
+###############################################################################
+# Show the performacne improvement when using a PointSet
+#
+# This is only available with VTK >= 9.1.0
+
+# pset = lidar.cast_to_pointset(deep=False)
+
+if pv.vtk_version_info >= (9, 1):
+    lidar_pset = lidar.cast_to_pointset()
+    tstart = time.time()
+    clipped_pset = lidar_pset.clip(origin=(0, 0, 1.76e3), normal=(0, 0, 1))
+    t_elapsed = time.time() - tstart
+    print(f"Time to clip with a PointSet {t_elapsed:.2f} seconds.")
+
+###############################################################################
+# Plot the same dataset
+#
+# .. note::
+#    PyVista must still create an intermediate PolyData to be able to plot, so
+#    there is no performance improvement when using a :class:`pyvista.PointSet`
+
+if pv.vtk_version_info >= (9, 1):
+    clipped_pset.plot()
```

Comment:
Should we address why the scalars are gone/inactivated in this version?

PolyData version:
![sphx_glr_create-pointset_001](https://user-images.githubusercontent.com/17914410/160385900-0c53e6c3-2976-4789-b9d6-f0b5671650b9.png)

PointSet version:
![sphx_glr_create-pointset_002](https://user-images.githubusercontent.com/17914410/160385906-ece2a7b3-a2da-4f69-bf8b-a052169d3129.png)